packages/orchestration/src/orchestration-client.ts

```diff
@@ -36,7 +36,11 @@ export class OrchestrationClient {
     prompt?: Prompt,
     requestConfig?: CustomRequestConfig
   ): Promise<OrchestrationResponse> {
-    const body = constructCompletionPostRequest(this.config, prompt);
+    const body: CompletionPostRequest | Record<string, any> =
```

Comment:
[req] You don't need to cast body to `Record<string, any>`. If you parse the JSON string, it will essentially be of type `CompletionPostRequest`, you can just add the prompt to it right?

Reply:
Yes but I was also considering the scenario where perhaps a spec change adds additional parameters which have not been adopted yet in our SDK. Since we are not checking for type safety, it can possibly provide users with a way to run their configurations still. Any thoughts on this?

Reply:
> provide users with a way to run their configurations

Yes, this is one of the use cases we want to cover with this approach

Reply:
Technically, adding Record or not won't have any effect on user, because this body is just a local variable, which will never affect any user. 

I would also remove `Record<string, any>` unless we want to be 100% strictly correct and say the JSON object parsed from the string can have breaking changes. 

No strong opinion and I see both points. In the end the type for body does not matter, and return type from `constructCompletionFromJson` is just for our own concern. The method is internal.

Reply:
That is correct. However, I still feel like it reads well to have it since it is explicitly clear to a technical user that the object can contain properties aside from those in `CompletionPostRequest`.  Plus, removing it here would also mean that the return type for `constructCompletionFromJson` is also of type `CompletionPostRequest` which, strictly speaking, might not be the case.

Reply:
I am also fine if you remove the type from here completely. Because:

1. Removing `Record<string, any>` is a misrepresentation of the type of `body`, since we are not enforcing adherence of type returned from `constructCompletionFromJson`  to `CompletionPostRequest`
2. `body` is not used after this point for any local operations for which type-safety might be needed. Also, the data property of execute function is `any` which makes the union typing here moot.

I would maybe document that body can have additional types for clarity

Reply:
Just for documentation, since we now need to be able to dynamically add streaming options, we actually do need some degree of type safety 😬